### ISSUE TITLE
[BUG] Inconsistent batch size output

### ISSUE THREAD

**Issue by andraugust** [kornia/kornia#460]
Thanks for the great package.  The output of `kornia.augmentation.RandomRotation` has an inconsistent batch size when the input batch size is 1:

```python
import torch
from kornia.augmentation import RandomRotation  # version 0.2.0

rotation = RandomRotation((-90.,90.))

x = torch.rand(1,3,10,10)
print(rotation(x).shape)   # torch.Size([3, 10, 10])  // expecting torch.Size([1, 3, 10, 10])

x = torch.rand(2,3,10,10)
print(rotation(x).shape)   # torch.Size([2, 3, 10, 10]) // fine

x = torch.rand(6,3,10,10)
print(rotation(x).shape)   # torch.Size([6, 3, 10, 10]) // fine
```
Is this desired behavior?

**Comment by skmatz** [comment]
The same behavior occurs not only with `RandomRotation`, but also with `RandomHorizontalFlip`, but not with `RandomVerticalFlip`.

Given this, this behavior seems to be inconsistent.
In the case of `RandomRotation`, the shape of the tensor changes around here:

https://github.com/kornia/kornia/blob/cb9373bbe0b98103ad982b9a2317fe472dd2e631/kornia/augmentation/functional.py#L487-L488

If this is unexpected behavior, we need to change the implementation.

**Comment by edgarriba** [comment]
@skmatz @andraugust thanks for reporting. Could you try to send a PR fixing ?

**Comment by skmatz** [comment]
@edgarriba

Yup, but first let's discuss (tell me) input/output shape.

| # | input shape    | output shape                              |
|---|----------------|-------------------------------------------|
| 1 | (1, 3, 10, 10) | (1, 3, 10, 10) or (3, 10, 10)             |
| 2 | (2, 3, 10, 10) | (2, 3, 10, 10)                            |
| 3 | (1, 1, 10, 10) | (1, 1, 10, 10) or (1, 10, 10) or (10, 10) |
| 4 | (2, 1, 10, 10) | (2, 1, 10, 10) or (2, 10, 10)             |
| 5 | (1, 10, 10)    | (1, 1, 10, 10) or (1, 10, 10) or (10, 10) |
| 6 | (3, 10, 10)    | (1, 3, 10, 10) or (3, 10, 10)             |

I would also like to discuss the form of acceptable input.
Whether to allow grayscale images (#_3, #_4) and non-batch images (#_5, #_6).

I don't think it is necessary to allow input non-batch images (#_5, #_6).
By allowing only (B, C, H, W) input shape, the output shape can be uniquely unified to (B, C, H, W).

What do you think?

**Comment by andraugust** [comment]
> By allowing only (B, C, H, W) input shape, the output shape can be uniquely unified to (B, C, H, W).

I agree. I think input and output shape should always be (B, C, H, W), even when B=1 and/or C=1.

**Comment by ducha-aiki** [comment]
I think that keeping (B, C, H, W) even for B=1 and/or C=1 is good convention to enforce.

**Comment by edgarriba** [comment]
@skmatz if you can proceed with some test and fixes for shape it's fine

**Comment by shijianjian** [comment]
It should be all fixed in #514, all the input will be transformed into BCHW and output BCHW only:
```python
def _transform_input(input: torch.Tensor) -> torch.Tensor:
    r"""Reshape an input tensor to be (*, C, H, W). Accept either (H, W), (C, H, W) or (*, C, H, W).
    Args:
        input: torch.Tensor

    Returns:
        torch.Tensor
    """
    if not torch.is_tensor(input):
        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")

    if len(input.shape) not in [2, 3, 4]:
        raise ValueError(
            f"Input size must have a shape of either (H, W), (C, H, W) or (*, C, H, W). Got {input.shape}")

    if len(input.shape) == 2:
        input = input.unsqueeze(0)

    if len(input.shape) == 3:
        input = input.unsqueeze(0)

    return input
```
